circus-train-s3-s3-copier/src/main/java/com/hotels/bdp/circustrain/s3s3copier/aws/JceksAmazonS3ClientFactory.java

```diff
@@ -103,13 +103,27 @@ private String regionForUri(AmazonS3 client, AmazonS3URI uri) {
     return bucketRegion;
   }
 
+  private AmazonS3ClientBuilder applyClientConfigurations(AmazonS3ClientBuilder builder, S3S3CopierOptions s3s3CopierOptions) {
+    ClientConfiguration clientConfiguration = new ClientConfiguration();
+
+    if (s3s3CopierOptions.getMaxThreadPoolSize() != -1) {
```

Comment:
Depending on what you decide to do based on @teabot's comment about the -1 magic number, it might be worth introducing a "can't be <=0 check" when we get the config and then this can just check >0

Reply:
removed entirely as per @patduin comments